File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/NewsArticle.java
package com.eulersbridge.isegoria.network.api.models;

import android.support.annotation.Nullable;

import com.eulersbridge.isegoria.network.adapters.Timestamp;
import com.squareup.moshi.Json;

import java.util.List;

@org.parceler.Parcel
public class NewsArticle {

    @Json(name = "articleId")
    public long id;

    @Nullable
    public Long institutionId;

    public String title;
    public String content;

    List<Photo> photos;

    @Json(name = "date")
    @Timestamp
    public long dateTimestamp;

    @Json(name = "likes")
    public int likeCount = 0;

    @Json(name = "creatorProfile")
    public Contact creator;

    @Json(name = "inappropriateContent")
    public boolean hasInappropriateContent;

    public @Nullable String getPhotoUrl() {
        if (photos != null && photos.size() > 0) {
            return photos.get(0).thumbnailUrl;

        } else {
            return null;
        }
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/NewsArticle.kt
package com.eulersbridge.isegoria.network.api.models

import android.annotation.SuppressLint
import android.os.Parcelable
import com.eulersbridge.isegoria.network.adapters.Timestamp
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@SuppressLint("ParcelCreator")
@Parcelize
data class NewsArticle(
        @Json(name = "articleId")
        val id: Long,

        val institutionId: Long?,
        val title: String,
        val content: String,
        val photos: List<Photo>?,

        @Timestamp
        val date: Long,

        @Json(name = "likes")
        val likeCount: Int = 0,

        @Json(name = "creatorProfile")
        val creator: Contact,

        @Json(name = "inappropriateContent")
        val hasInappropriateContent: Boolean

) : Parcelable {
    internal fun getPhotoUrl(): String? {
        return photos?.firstOrNull()?.thumbnailUrl
    }
}